### PR TITLE
Vue 3x - Update for titleId and maskId prop

### PR DESCRIPTION
**Pull request by jasonlundien**
This PR addresses this bug report:  [<font-awesome-icon> title property breaks snapshots#181](https://github.com/FortAwesome/vue-fontawesome/issues/181)

Currently we cannot add and set a title property without a random generated string being attached to the `aria-labelledby` and the `title id`.

image

This PR will allow us to set a `title` and `titleId` so that there would NOT be a random generated string if set, and snapshot tests will now pass.

image

@robmadole --- can you review code for accuracy?

### CODE REVIEW

File: src/components/FontAwesomeIcon.js [hunk 1]
```diff
@@ -1,167 +1,224 @@
...
+      default: false,
     },
     title: {
       type: String,
-      default: null
+      default: null,
+    },
+    titleId: {
+      type: String,
+      default: null,
```
Comment (jasonlundien) [inline src/components/FontAwesomeIcon.js]: added `titleId`

File: src/components/FontAwesomeIcon.js [hunk 2]
```diff
@@ -1,167 +1,224 @@
...
+      required: true,
     },
     mask: {
       type: [Object, Array, String],
-      default: null
+      default: null,
+    },
+    maskId: {
+      type: String,
+      default: null,
```
Comment (jasonlundien) [inline src/components/FontAwesomeIcon.js]: added `maskId`

File: index.js [hunk 3]
```diff
@@ -540,23 +523,25 @@
...
-        return objectWithKey('mask', normalizeIconArgs(props.mask));
+        return objectWithKey("mask", normalizeIconArgs(props.mask));
       });
       var renderedIcon = vue.computed(function () {
         return fontawesomeSvgCore.icon(icon.value, _objectSpread2(_objectSpread2(_objectSpread2(_objectSpread2({}, classes.value), transform.value), mask.value), {}, {
           symbol: props.symbol,
-          title: props.title
+          title: props.title,
+          titleId: props.titleId,
+          maskId: props.maskId
```
Comment (jasonlundien) [inline index.js:539]: added `titleId` and `maskId` here.

File: index.d.ts [hunk 4]
```diff
@@ -1,43 +1,61 @@
...
-  fade?: boolean
-  beatFade?: boolean
-  spinPulse?: boolean
-  spinReverse?: boolean
+  border?: boolean;
+  fixedWidth?: boolean;
+  flip?: "horizontal" | "vertical" | "both";
+  icon: object | Array<string> | string | IconDefinition;
+  mask?: object | Array<string> | string;
+  maskId?: object | Array<string> | string;
```
Comment (jasonlundien) [inline index.d.ts:10]: added `maskId` here.

File: index.d.ts [hunk 5]
```diff
@@ -1,43 +1,61 @@
...
+    | "6x"
+    | "7x"
+    | "8x"
+    | "9x"
+    | "10x";
+  spin?: boolean;
+  transform?: object | string;
+  symbol?: boolean | string;
+  title?: string;
+  titleId?: string;
```
Comment (jasonlundien) [inline index.d.ts:37]: added `titleId` here.

**Comment by jasonlundien** [comment]
Prettier has taken over these files as well @robmadole. I left comments where I changed the code (6 total).

- `index.es.js` and `index.js` were updated automatically with `npm run dist`
-  I did have to manually update `index.d.ts`, which is 1 of the 6 comments.

File: src/components/__tests__/FontAwesomeIcon.test.js [hunk 6]
```diff
@@ -2,454 +2,539 @@
...
-  test('using array format, short prefix and short icon name', () => {
-    const wrapper = mountFromProps({ icon: ['fas', 'coffee'] })
-
-    expect(wrapper.element.tagName).toBe('svg')
-    expect(wrapper.element.classList.contains('fa-coffee')).toBeTruthy()
-  })
+  library.reset();
+});
+
+describe("icon title prop", () => {
```
Comment (jasonlundien) [inline src/components/__tests__/FontAwesomeIcon.test.js:26]: This `describe` (originally, lines 26-55) block are the only tests I added.  
All other changes are prettier.

**Comment by robmadole** [comment]
@jasonlundien the changes look good. But the stuff prettier did is a little annoying. Were you surprised by them? I almost wonder if something is misconfigured in VSCode.

**Comment by jasonlundien** [comment]
@robmadole ---

> @jasonlundien the changes look good. But the stuff prettier did is a little annoying. Were you surprised by them? I almost wonder if something is misconfigured in VSCode.

I went back and adjusted my prettier settings to reflect those of Font Awesome.  The prettier changes are now fewer than they were but you will still see some the changes --- for consistency.  The exception is the `index.es.js` file which prettier did quite a number on... this file was auto updated when I ran the `npm run dist`.

I am fine with these prettier changes and being consistent going forward if you are ?!

File: src/components/__tests__/FontAwesomeIcon.test.js [hunk 7]
```diff
@@ -282,11 +305,13 @@ test('swap opacity', () => {
 })
 
 test('using size', () => {
-  ;['2xs', 'xs', 'sm', 'lg', 'xl', '2xl', '1x', '2x', '3x', '4x', '5x', '6x', '7x', '8x', '9x', '10x'].forEach((size) => {
-    const wrapper = mountFromProps({ icon: faCoffee, size: size })
+  ;['2xs', 'xs', 'sm', 'lg', 'xl', '2xl', '1x', '2x', '3x', '4x', '5x', '6x', '7x', '8x', '9x', '10x'].forEach(
```
Comment (jasonlundien) [inline src/components/__tests__/FontAwesomeIcon.test.js:308]: prettier

File: src/components/__tests__/FontAwesomeIcon.test.js [hunk 8]
```diff
@@ -282,11 +305,13 @@ test('swap opacity', () => {
...
+    (size) => {
+      const wrapper = mountFromProps({ icon: faCoffee, size: size })
 
-    expect(wrapper.element.classList.contains(`fa-${size}`)).toBeTruthy()
-  })
+      expect(wrapper.element.classList.contains(`fa-${size}`)).toBeTruthy()
```
Comment (jasonlundien) [inline src/components/__tests__/FontAwesomeIcon.test.js:312]: prettier

File: src/components/__tests__/FontAwesomeIcon.test.js [hunk 9]
```diff
@@ -282,11 +305,13 @@ test('swap opacity', () => {
 })
 
 test('using size', () => {
-  ;['2xs', 'xs', 'sm', 'lg', 'xl', '2xl', '1x', '2x', '3x', '4x', '5x', '6x', '7x', '8x', '9x', '10x'].forEach((size) => {
-    const wrapper = mountFromProps({ icon: faCoffee, size: size })
+  ;['2xs', 'xs', 'sm', 'lg', 'xl', '2xl', '1x', '2x', '3x', '4x', '5x', '6x', '7x', '8x', '9x', '10x'].forEach(
+    (size) => {
+      const wrapper = mountFromProps({ icon: faCoffee, size: size })
```
Comment (jasonlundien) [inline src/components/__tests__/FontAwesomeIcon.test.js]: prettier

File: src/components/FontAwesomeIcon.js [hunk 10]
```diff
@@ -133,35 +141,38 @@ export default defineComponent({
...
-    }, { immediate: true })
+    watch(
+      renderedIcon,
+      (value) => {
+        if (!value) {
+          return log('Could not find one or more icon(s)', icon.value, mask.value)
+        }
+      },
+      { immediate: true }
+    )
```
Comment (jasonlundien) [inline src/components/FontAwesomeIcon.js:173]: prettier

File: src/components/FontAwesomeIcon.js [hunk 11]
```diff
@@ -133,35 +141,38 @@ export default defineComponent({
...
 
-    const vnode = computed(() => renderedIcon.value ? convert(renderedIcon.value.abstract[0], {}, attrs) : null)
+    const vnode = computed(() => (renderedIcon.value ? convert(renderedIcon.value.abstract[0], {}, attrs) : null))
```
Comment (jasonlundien) [inline src/components/FontAwesomeIcon.js:175]: prettier

File: src/components/FontAwesomeIcon.js [hunk 12]
```diff
@@ -133,35 +141,38 @@ export default defineComponent({
...
+    const renderedIcon = computed(() =>
+      faIcon(icon.value, {
+        ...classes.value,
+        ...transform.value,
+        ...mask.value,
+        symbol: props.symbol,
+        title: props.title,
+        titleId: props.titleId,
+        maskId: props.maskId
+      })
```
Comment (jasonlundien) [inline src/components/FontAwesomeIcon.js:162]: prettier + added  `titleId: props.titleId,` and `maskId: props.maskId`

File: src/components/FontAwesomeIcon.js [hunk 13]
```diff
@@ -133,35 +141,38 @@ export default defineComponent({
...
+  setup(props, { attrs }) {
     const icon = computed(() => normalizeIconArgs(props.icon))
     const classes = computed(() => objectWithKey('classes', classList(props)))
-    const transform = computed(() => objectWithKey(
-      'transform',
-      (typeof props.transform === 'string')
-        ? faParse.transform(props.transform)
-        : props.transform
-    ))
+    const transform = computed(() => objectWithKey('transform', typeof props.transform === 'string' ? faParse.transform(props.transform) : props.transform))
```
Comment (jasonlundien) [inline src/components/FontAwesomeIcon.js:150]: prettier

File: src/components/FontAwesomeIcon.js [hunk 14]
```diff
@@ -133,35 +141,38 @@ export default defineComponent({
     spinReverse: {
       type: Boolean,
       default: false
-    },
+    }
   },
 
-  setup (props, { attrs }) {
+  setup(props, { attrs }) {
```
Comment (jasonlundien) [inline src/components/FontAwesomeIcon.js:147]: prettier

File: src/components/FontAwesomeIcon.js [hunk 15]
```diff
@@ -133,35 +141,38 @@ export default defineComponent({
     spinReverse: {
       type: Boolean,
       default: false
-    },
+    }
```
Comment (jasonlundien) [inline src/components/FontAwesomeIcon.js:144]: prettier

File: src/components/FontAwesomeIcon.js [hunk 16]
```diff
@@ -4,7 +4,7 @@ import convert from '../converter'
 import log from '../logger'
 import { objectWithKey, classList } from '../utils'
 
-function normalizeIconArgs (icon) {
+function normalizeIconArgs(icon) {
```
Comment (jasonlundien) [inline src/components/FontAwesomeIcon.js:7]: prettier

**Pull request merged**